app/plugin_install.go

```diff
@@ -17,12 +17,29 @@ import (
 	"github.com/mattermost/mattermost-server/utils"
 )
 
+const managedPluginFileName = ".filestore"
+const fileStorePluginFolder = "./plugins"
+
 func (a *App) InstallPluginFromData(data model.PluginEventData) {
 	mlog.Info(fmt.Sprintf("InstallPluginFromData. ID: %v, Path: %v", data.Id, data.FileStorePath))
+
+	reader, appErr := a.FileReader(data.FileStorePath)
+	if appErr != nil {
+		mlog.Error("Failed to open plugin bundle from filestore.", mlog.String("bundle", data.FileStorePath), mlog.Err(appErr))
+	}
+	defer reader.Close()
+
+	if _, appErr = a.installPluginLocally(reader, true); appErr != nil {
+		mlog.Error("Failed to unpack plugin from filestore", mlog.Err(appErr), mlog.String("path", data.FileStorePath))
+	}
 }
 
 func (a *App) RemovePluginFromData(data model.PluginEventData) {
 	mlog.Info(fmt.Sprintf("RemovePluginFromData. ID: %v, Path: %v", data.Id, data.FileStorePath))
```

Comment:
This can be removed from here, since I only added it for testing purposes.

Reply:
✅Demoted to mlog.Debug per conversation above